File: 287. Find the Duplicate Number/AC_Answer_2.py
# Use O(1) space, O(n) time, quicker than previous answer

class Solution(object):
    def findDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        nums.sort()
        count, temp = 1, nums[0]
        for i in range(1, len(nums)):
            if nums[i] == temp:
                count += 1
            else:
                temp = nums[i]
                count = 1
            if count > 1:
                return temp

File: 434. Number of Segments in a String/AC_Answer_2.java
class Solution {
    public int countSegments(String s) {
        if(s.isEmpty()) return 0;
        int i=0, count = 0;
        for(int j=0; j<s.length(); j++){
            if(s.charAt(j)==' ' || j==s.length()-1){
                if(s.charAt(i) !=' '){
                    count++;
                }
                i = j+1;
            }
        }
        return count;
    }
}
